iceoryx_hoofs/include/iceoryx_hoofs/internal/concurrent/lockfree_queue/index_queue.hpp

```diff
@@ -116,6 +116,7 @@ class IndexQueue
     ///    the default atomic constructor does not call the default constructor of the
     ///    underlying class.
     ///    See, http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2018/p0883r0.pdf
+    // NOLINTNEXTLINE(cppcoreguidelines-avoid-c-arrays, hicpp-avoid-c-arrays)
     Cell m_cells[Capacity];
```

Comment:
Couldn't the `buffer` be used for this? Not in this PR but just for a future refactoring?

Reply:
Better use some initialized array here as we want the ctor to be called for each element. Maybe even std::array.

Reply:
Changed to `std::array` (which does not allocate nor throw if used correctly (only in bounds checked access).

Reply:
@MatthiasKillat oh, then this was a misunderstanding on my side. I assumed `Cell` to be an alias to something like `uint8_t[N]`. The C array would have been okay here.

Reply:
`using Cell = std::atomic<Index>;` is an atomic of lightweight structure, but not a primitive type.

Do we not want to use `std::array` or not? I cannot help but think that our coding or design guidelines combined with the clang-tidy rules are not ideal in this case. We unnecessarily restrict everything and additionally make the code more verbose/ugly.

It is also at odds with the clang tidy rules in this case (which says use something like a std::array). Now we do not have this something of our own yet (and I would argue we do not need it as std::array is fine for us as well).

Reply:
Except that we removed the `std::array` in favor of the C-array to keep the STL dependency low, there is no technical reason not to use it. I would keep the C-array for now and maybe we create a small `cxx::array` wrapper for the `buffer`/`uninitialized_buffer`